mne/epochs.py

```diff
@@ -834,33 +839,48 @@ def _reject_setup(self, reject, flat):
             "previous ones"
         )
 
+        # Skip this check if old_reject, reject, old_flat, and flat are
+        # callables
+        is_callable = False
+        for rej in (reject, flat, old_reject, old_flat):
+            for key, val in rej.items():
+                if callable(val):
+                    is_callable = True
+
         # copy thresholds for channel types that were used previously, but not
         # passed this time
         for key in set(old_reject) - set(reject):
             reject[key] = old_reject[key]
-        # make sure new thresholds are at least as stringent as the old ones
-        for key in reject:
-            if key in old_reject and reject[key] > old_reject[key]:
-                raise ValueError(
-                    bad_msg.format(
-                        kind="reject",
-                        key=key,
-                        new=reject[key],
-                        old=old_reject[key],
-                        op=">",
+
+        if not is_callable:
+            # make sure new thresholds are at least as stringent
+            # as the old ones
```

Comment:
... but I think you can make these checks more fine-grained without much more work. Instead of if *any* are callable, skip *all* checks (which is what you're doing now), you should within this loop be able to fairly easily make it so that if *this* type is or was callable, skip *this* check. I think it almost amounts to moving the conditional from above into this loop with a `continue` when callables are or were present. As a bonus, your `diff` will probably be smaller, too!

Reply:
Should I edit my test to check for this?